upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1429,8 +1425,8 @@ func (n *nodeUpConfigBuilder) BuildConfig(ig *kops.InstanceGroup, apiserverAddit
 	useConfigServer := apiModel.UseKopsControllerForNodeBootstrap(cluster) && !ig.HasAPIServer()
 	if useConfigServer {
 		host := "kops-controller.internal." + cluster.ObjectMeta.Name
-		if cluster.UsesNoneDNS() {
-			host = bootConfig.APIServerIP
+		if cluster.UsesNoneDNS() && len(bootConfig.APIServerIPs) > 0 {
+			host = bootConfig.APIServerIPs[0] // TODO: how we could support array?
```

Comment:
how to deal with this? or is one ip enough?

Reply:
Let's discuss this separately.